lib/instrumentation/modules/apollo-server-core.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+var semver = require('semver')
+
+module.exports = function (apolloServerCore, agent, version, enabled) {
+  if (!enabled) return apolloServerCore
+
+  if (!semver.satisfies(version, '^2.0.2') || typeof apolloServerCore.runHttpQuery !== 'function') {
+    agent.logger.debug('apollo-server-core version %s not supported - aborting...', version)
+    return apolloServerCore
+  }
+
+  var runHttpQuery = apolloServerCore.runHttpQuery
+  apolloServerCore.runHttpQuery = wrappedRunHttpQuery
```

Comment:
This could also be written like this:

```js
shimmer.wrap(apolloServerCore, 'runHttpQuery', function (runHttpQuery) {
  return function wrappedRunHttpQuery () {
    var trans = agent._instrumentation.currentTransaction
    if (trans) trans._graphqlRoute = true
    return runHttpQuery.apply(this, arguments)
  }
})
```

Just add `const shimmer = require('../shimmer')` to the requires at the top.

The benefit is that shimmer provides the checking so you don't have to make sure `runHttpQuery` is a function before trying to patch it.